File: server/models/Service.js
const mongoose = require('mongoose');

const serviceSchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Tên dịch vụ là bắt buộc'],
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  shortDescription: {
    type: String,
    trim: true
  },
  price: {
    type: Number,
    required: [true, 'Giá dịch vụ là bắt buộc'],
    min: 0
  },
  specialtyId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Specialty',
    required: [true, 'Chuyên khoa là bắt buộc']
  },
  duration: {
    type: Number,
    default: 30, // Thời gian mặc định: 30 phút
    min: 10
  },
  type: {
    type: String,
    enum: ['examination', 'diagnostic', 'treatment', 'procedure', 'surgery', 'consultation'],
    default: 'examination'
  },
  preparationGuide: {
    type: String,
    trim: true
  },
  aftercareInstructions: {
    type: String,
    trim: true
  },
  requiredTests: [{
    type: String,
    trim: true
  }],
  isActive: {
    type: Boolean,
    default: true
  },
  image: {
    url: String,
    secureUrl: String,
    publicId: String,
    cloudName: String
  },
  imageUrl: {
    type: String,
    trim: true
  }
}, {
  timestamps: true
});

// Index for faster queries
serviceSchema.index({ name: 1 });
serviceSchema.index({ specialtyId: 1 });
serviceSchema.index({ price: 1 });
serviceSchema.index({ isActive: 1 });
serviceSchema.index({ type: 1 });

module.exports = mongoose.model('Service', serviceSchema); 

File: hospital_mobile_app/lib/domain/entities/service.dart
import 'package:equatable/equatable.dart';

class Service extends Equatable {
  final String id;
  final String name;
  final String description;
  final double price;
  final String? image;
  final String? specialtyId;
  final String? specialtyName;
  final String? shortDescription;
  final int? duration;
  final String? type;
  final String? instructions;
  final String? preparationGuide;
  final String? aftercareInstructions;
  final List<String>? requiredTests;
  final DateTime createdAt;

  const Service({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    this.image,
    this.specialtyId,
    this.specialtyName,
    this.shortDescription,
    this.duration,
    this.type,
    this.instructions,
    this.preparationGuide,
    this.aftercareInstructions,
    this.requiredTests,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        price,
        image,
        specialtyId,
        specialtyName,
        shortDescription,
        duration,
        type,
        instructions,
        preparationGuide,
        aftercareInstructions,
        requiredTests,
        createdAt,
      ];
}
